File: Daily Challanges/2026-05-May/2026-05-10.py
def is_valid_isbn_13(s):
    digit_sum = 0
    digit_count = 0
    for char in s:
        if char == "-":
            continue
        if not char.isdigit():
            print("not digit")
            return False
        digit_count += 1
        digit_sum += int(char) * (1 if digit_count % 2 == 1 else 3)
    return digit_count == 13 and digit_sum % 10 == 0


print(is_valid_isbn_13("9780306406157"))
print(is_valid_isbn_13("97803064061570"))
print(is_valid_isbn_13("978-0-13-595705-9"))

File: Daily Challanges/2026-05-May/2026-05-10.js
function isValidIsbn13(str) {
  let sum = 0;
  let digit_count = 0;
  let is_even = false;
  for (const char of str) {
    if (char === "-") continue;
    if (!/[0-9]/.test(char)) return false;
    sum += Number(char) * (is_even ? 3 : 1);
    is_even = !is_even;
    digit_count += 1;
  }
  return digit_count === 13 && sum % 10 === 0;
}

console.log(isValidIsbn13("9780306406157"));
console.log(isValidIsbn13("97803064061570"));
